hathor/p2p/peer.py

```diff
@@ -11,13 +11,36 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""
+This module exposes three peer classes that share similar behavior but must not be mixed.
+
+This is the class inheritance structure:
+
+    _Peer → Peer
+      ↓
+    _PublicPeer → PublicPeer
+      ↓
+    _PrivatePeer → PrivatePeer
+
+This way the shared behavior is implemented and propagated through the private classes, and the public classes don't
+share the same inheritance tree and for example a `peer: PublicPeer` will have `isinstance(peer, Peer) == False`, so
+they can't be mixed.
+
+This makes it harder for external functions to support "subtypes" by accepting a base class, but this is intentional.
+If a function can work for any type of peer, it should be defined as `def foo(peer: Peer)` and callers will have to
+call `private_peer.to_peer()` because `PrivatePeer` is not a subclass of `Peer`.
+"""
```

Comment:
Wouldn't it be simpler to use composition over inheritance? You could have independent `PublicPeer` and `PrivatePeer` dataclasses, with a `Peer` attribute in each encompassing shared behavior.

Reply:
Done.

Reply:
Great, thanks!